pkg/motor/interface.go

```diff
@@ -0,0 +1,44 @@
+package motor
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/sonr-io/sonr/internal/bucket"
+	"github.com/sonr-io/sonr/pkg/client"
+	"github.com/sonr-io/sonr/pkg/crypto/mpc"
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/pkg/host"
+	mt "github.com/sonr-io/sonr/pkg/motor/types"
+	"github.com/sonr-io/sonr/pkg/motor/x/object"
+	bt "github.com/sonr-io/sonr/x/bucket/types"
+)
+
+type MotorNode interface {
+	GetDeviceID() string
+
+	GetAddress() string
+	GetBalance() int64
+
+	GetClient() *client.Client
+	GetWallet() *mpc.Wallet
+	GetPubKey() *secp256k1.PubKey
```

Comment:
Are we sure we want to expose a specific Key type here? It seems like it will be difficult to change  in the future.

Reply:
these are valid points on the current `motor` interface. but out of scope of the current pr

Reply:
@SCKelemen open an issue for this. May want to fix this soonish. Do you have a better idea of how to handle it?